packages/fxa-auth-server/lib/oauth/jwt_id_token.js

```diff
@@ -97,10 +110,17 @@ exports.verify = async function verify(idToken, clientId) {
   // 9. The current time MUST be before the time represented by the `exp`
   //    Claim.
   //
+  // Note that we optionally allow for expired tokens via the
+  // `expiryGracePeriod` argument.
+  //
   // Note that this time is specified to be in seconds, not milliseconds.
   const currentTime = Math.round(Date.now() / 1000);
-  if (claims.exp < currentTime) {
-    logger.debug('verify.error.exp', { exp: claims.exp, currentTime });
+  if (claims.exp + expiryGracePeriod < currentTime) {
```

Comment:
Be careful here, the `ignoreExpiration` flag causes jsonwebtoken to skip *all* validation of the `exp` field, including checking whether it's a number. That's probably OK since we've already checked the signature and we'd never issue a malformed JWT, but it's worth being aware of.

Reply:
ooh, good point, I had missed that. I'll add a check that the 'exp' claim value is present, and is a number no more than 5 minutes in the future.